File: src/cameo/ResultEvent.cpp
/*
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under the EUPL, Version 1.1 only (the "License");
 * You may not use this work except in compliance with the Licence.
 * You may obtain a copy of the Licence at:
 *
 * http://joinup.ec.europa.eu/software/page/eupl
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the Licence for the specific language governing permissions and
 * limitations under the Licence.
 */

#include "../cameo/ResultEvent.h"

#include <iostream>

namespace cameo {

ResultEvent::ResultEvent(int id, const std::string& name, const std::string& data) :
	Event(id, name),
	m_data(data) {
}

const std::string& ResultEvent::getData() const {
	return m_data;
}

std::ostream& operator<<(std::ostream& os, const cameo::ResultEvent& status) {
	os << "name=" << status.m_name
		<< "\nid=" << status.m_id;

	return os;
}

}

File: java/api/src/main/java/fr/ill/ics/cameo/base/ResultEvent.java
/*
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under the EUPL, Version 1.1 only (the "License");
 * You may not use this work except in compliance with the Licence.
 * You may obtain a copy of the Licence at:
 *
 * http://joinup.ec.europa.eu/software/page/eupl
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the Licence is distributed on an "AS IS" basis,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the Licence for the specific language governing permissions and
 * limitations under the Licence.
 */

package fr.ill.ics.cameo.base;

import org.json.simple.JSONObject;

/**
 * Class defining a result event.
 */
public class ResultEvent extends Event {
	
	private byte[] data;
	
	/**
	 * Constructor.
	 * @param id The application id.
	 * @param name The application name.
	 * @param data The result data.
	 */
	public ResultEvent(int id, String name, byte[] data) {
		super(id, name);
		this.data = data;
	}
		
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		ResultEvent other = (ResultEvent) obj;
		
		if (id != other.id) {
			return false;
		}
		if (name == null) {
			if (other.name != null) {
				return false;
			}
		} else if (!name.equals(other.name)) {
			return false;
		}
		return true;
	}

	public byte[] getData() {
		return data;
	}
	
	@Override
	public String toString() {
		JSONObject result = new JSONObject();
		
		result.put("type", "result");
		result.put("id", id);
		result.put("name", name);
		
		return result.toJSONString();
	}
	
}